test/Microsoft.VisualStudio.Editor.Razor.Test/BraceSmartIndenterTest.cs

```diff
@@ -0,0 +1,382 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.Editor.Razor.Test.Infrastructure;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+using Moq;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    public class BraceSmartIndenterTest : ForegroundDispatcherTestBase
+    {
+        [Fact]
+        public void TextBuffer_OnPostChanged_MovesToEndOfLine()
+        {
+            // Arrange
+            var change = Environment.NewLine;
+            var initialSnapshot = new StringTextSnapshot("@{ }");
+            var afterChangeSnapshot = new StringTextSnapshot("@{ " + change + "}");
+            var edit = new TestEdit(3, 0, initialSnapshot, change.Length, afterChangeSnapshot, change);
+            var expectedIndentResult = "@{ " + change + change + "}";
+
+            var bufferPosition = new VirtualSnapshotPoint(afterChangeSnapshot, 3 + change.Length);
+            var caret = CreateCaretFrom(3 + change.Length, afterChangeSnapshot);
+            TestTextBuffer textBuffer = null;
+            var focusedTextView = CreateFocusedTextView(() => textBuffer, caret);
+            textBuffer = CreateTrackedTextBuffer(initialSnapshot, focusedTextView);
+            var editorOperations = new Mock<IEditorOperations>();
+            editorOperations.Setup(operations => operations.MoveToEndOfLine(false));
+            var editorOperationsFactory = new Mock<IEditorOperationsFactoryService>();
+            editorOperationsFactory.Setup(factory => factory.GetEditorOperations(It.IsAny<ITextView>()))
+                .Returns(editorOperations.Object);
+            var braceSmartIndenter = new BraceSmartIndenter(Dispatcher, textBuffer, editorOperationsFactory.Object);
```

Comment:
This is really opaque and hard to understand, can we move some of this boilerplate to properties

Reply:
Why not have it opaque? It's a standalone component.